exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/JoinUtils.java

```diff
@@ -231,6 +236,12 @@ public static boolean isScalarSubquery(RelNode root) {
         return true;
       }
     }
+    if (!hasMoreInputs && currentRel != null) {
+      Double rowSize = RelMetadataQuery.instance().getMaxRowCount(currentRel);
```

Comment:
Actually,  I am thinking this entire method isScalarSubquery() should be rewritten to call RelMetadataQuery.getMaxRowCount().   Originally, when this was written, the max row count interface was not available. 
I think your use case is somewhat specific that the table has 1 row.  However, suppose we had LIMIT 1 in the subquery, we could still treat it as scalar since it would produce at most 1 row.  My thought is instead of solving for your specific table scan case, we might as well address the generalized scalar check since Calcite already provides it.

Reply:
I agree that we should rewrite the method by call RelMetadataQuery.getMaxRowCount(). btw, could you give a guide how calcite do that ?

Reply:
You could see the usage of RelMdMaxRowCount in Calcite [1].  It has implementations for different Rels including Aggregate, sort+limit.  I would recommend adding some unit tests though if you are going to make the code changes since this would be new functionality.  

[1] https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/rel/metadata/RelMdMaxRowCount.java